domain/relation/service/relation.go

```diff
@@ -268,7 +302,19 @@ func (s *WatchableService) WatchApplicationSettings(
 	relationUUID corerelation.UUID,
 	applicationID application.ID,
 ) (watcher.NotifyWatcher, error) {
-	return nil, errors.NotImplemented
+	relationEndpointUUID, err := s.GetRelationEndpointUUID(ctx, relation.GetRelationEndpointUUIDArgs{
+		RelationUUID:  relationUUID,
+		ApplicationID: applicationID,
+	})
+	if err != nil {
+		return nil, internalerrors.Capture(err)
+	}
+	return s.watcherFactory.NewNamespaceNotifyMapperWatcher(
+		"relation_application_setting",
```

Comment:
todo: Provide table name via a state method. The sql table details should stay contained in state.

See `InitialWatchStatementUnitLife()` and `WatchOpenedPortsTable()`